doc/function_types.md

```diff
@@ -1144,3 +1144,66 @@ including ensmallen:
 This can be useful for situations where you know that the checks should be
 ignored.  However, be aware that the code may fail to compile and give more
 confusing and difficult error messages!
+
+
+## Multi objective functions
```

Comment:
I think we should go with the spelling `multi-objective`, since that's what's used most places (and as far as I'm aware is  grammatically correct). There are a couple other places to fix this.

Reply:
Alright, I will make the changes! :)